sdk/rust/oak/src/grpc/mod.rs

```diff
@@ -191,9 +194,12 @@ where
     Q: protobuf::Message,
 {
     let r: R = protobuf::parse_from_bytes(&req).expect("Failed to parse request protobuf message");
-    match node_fn(r) {
+    let result = match node_fn(r) {
         Ok(rsp) => writer.write(&rsp, WriteMode::Close),
         Err(status) => writer.close(Err(status)),
+    };
+    if let Err(e) = result {
+        panic!("Failed to process response: {}", e)
     }
```

Comment:
Could this be something like

```rust
node_fn(r)
    .and_then(|rsp| writer.write(&rsp, WriteMode::Close))
    .or_else(|status| writer.close(Err(status))
    .unwrap_or_else(|e| panic!("Failed to process response: {}", e));
```

If the closures get in the way though it's probably not worth trying to contort the code too much, and it would be better to leave as it is.

Reply:
Or, better, leave the last handling to the caller by making the function return a `Result`? If not, could you add a comment to clarify that / when this function panics?

Reply:
Added doc comments for the moment.